src/GraphClient.ts

```diff
@@ -50,25 +55,68 @@ export default class GraphClient {
     // GraphQL query that is parsed into the standard GraphQL AST(Abstract syntax tree)
     const gqlSubscriptionQry = gql`${subscriptionQry}`;
     // Subscription handling
-    const querySubscriber = await Promise.resolve(this.apolloClient.subscribe({
+    const observable = this.apolloClient.subscribe({
       query: gqlSubscriptionQry,
       variables: {},
-    }).subscribe({
-      async next(response) {
-        const transactions = await fetcher.fetch(response.data);
-        await handler.handle(transactions);
-        // Integrate updation of ContractEntity uts here. Handlers should make sure error
-        // Promise.reject is thrown on error cases.
-      },
-      error(err) {
-        // Log error using logger
-        Logger.error(err);
-      },
-    }));
+    });
+    const querySubscriber = await Promise.resolve(
+      observable
+        .subscribe({
+          async next(response) {
+            const transactions = await fetcher.fetch(response.data);
+            await handler.handle(transactions);
+            await GraphClient.updateLatestUTS(
+              transactions,
+              response.data,
+              contractEntityRepository,
+            );
+          },
+          error(err) {
+            Logger.error(err);
+          },
+        }),
+    );
 
     return querySubscriber;
   }
 
+  /**
+   * This method updates latest timestamp for contract entities.
+   * @param transactions Transactions for transaction fetcher.
+   * @param subscriptionResponse Subscription response.
+   * @param contractEntityRepository Instance of contract entity repository.
+   */
+  private static async updateLatestUTS(
+    transactions: Record<string, Record<string, any>>,
+    subscriptionResponse: Record<string, any[]>,
+    contractEntityRepository: ContractEntityRepository,
+  ): Promise<void> {
+    const savePromises = Object.keys(transactions).map(
+      async (transactionKind) => {
+        const { contractAddress } = subscriptionResponse[transactionKind][0];
+        const transaction = transactions[transactionKind].length > 0
+          ? transactions[transactionKind][transactions[transactionKind].length - 1]
+          : null;
+
+        const number = transaction ? transaction.uts : 0;
```

Comment:
I would rather not use `number` as a member variable. Also, transaction !== null looks to me better as a check condition.

```ts
        const currentUTS = new BigNumber(
          transaction !== null ? transaction.uts : 0
        );
```

Reply:
Also, fallback to 0 seems "strange" to me (as a UTC value). Is this an error situation?

Reply:
I will remove this check because this filed will always be received from `thegraph` node.